core/src/worker/workflow/mod.rs

```diff
@@ -475,6 +477,11 @@ impl Workflows {
                         if same_task_queue {
                             if let Some(p) = at_handle.reserve_slot() {
                                 reserved.push(p);
+                                if reserved.len()
+                                    >= MAX_EAGER_ACTIVITY_RESERVATIONS_PER_WORKFLOW_TASK
+                                {
+                                    break;
+                                }
```

Comment:
This won't work. Commands come in with `request_eager_execution` set true. You need to set it false for the remaining commands in the list.

A test should prove this works

Reply:
Oh, I forgot about that part, thanks